### PR TITLE
Userland: Add pgrep

### PR DESCRIPTION
**Pull request by abyesilyurt**

### CODE REVIEW

File: Userland/Utilities/pgrep.cpp [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+int main(int argc, char** argv)
+{
+    if (pledge("stdio rpath", nullptr) < 0) {
+        perror("pledge");
+        return 1;
+    }
+
+    bool case_insensitive = false;
+    bool invert_match = false;
+    const char* pattern = nullptr;
```
Comment (alimpfard) [inline Userland/Utilities/pgrep.cpp:22]: Please use east-const (as per the project code style)

File: Userland/Utilities/pgrep.cpp [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        auto result = re.match(it.name, PosixFlags::Global);
+        if (result.success ^ invert_match) {
+            matches.append(it.pid);
+        }
+    }
+
+    quick_sort(matches);
+
+    for (auto& it : matches) {
+        outln("{}", it);
```
Comment (alimpfard) [inline Userland/Utilities/pgrep.cpp:54]: This is not an iterator, so let's call it `match` or something similar instead.

File: Userland/Utilities/pgrep.cpp (hunk 1 above)
Comment (abyesilyurt) [reply Userland/Utilities/pgrep.cpp]: Done!

File: Userland/Utilities/pgrep.cpp (hunk 2 above)
Comment (abyesilyurt) [reply Userland/Utilities/pgrep.cpp]: Done!

**Pull request merged**